account_easy_reconcile/easy_reconcile.py

```diff
@@ -0,0 +1,424 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright 2012-2013 Camptocamp SA (Guewen Baconnier)
+#    Copyright (C) 2010   Sébastien Beau
+#    Copyright 2015 Camptocamp SA (Damien Crier)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from datetime import datetime
+from openerp import models, api, fields, _
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from openerp.exceptions import except_orm
+from openerp import sql_db
+
+# from openerp import pooler
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class EasyReconcileOptions(models.AbstractModel):
+    """Options of a reconciliation profile
+
+    Columns shared by the configuration of methods
+    and by the reconciliation wizards.
+    This allows decoupling of the methods and the
+    wizards and allows to launch the wizards alone
+    """
+
+    _name = 'easy.reconcile.options'
+
+    @api.model
+    def _get_rec_base_date(self):
+        return [
+            ('end_period_last_credit', 'End of period of most recent credit'),
+            ('newest', 'Most recent move line'),
+            ('actual', 'Today'),
+            ('end_period', 'End of period of most recent move line'),
+            ('newest_credit', 'Date of most recent credit'),
+            ('newest_debit', 'Date of most recent debit')
+        ]
+
+    write_off = fields.Float('Write off allowed', default=0.)
+    account_lost_id = fields.Many2one('account.account',
+                                      string="Account Lost")
+    account_profit_id = fields.Many2one('account.account',
+                                        string="Account Profit")
+    journal_id = fields.Many2one('account.journal',
+                                 string="Journal")
+    date_base_on = fields.Selection('_get_rec_base_date',
+                                    required=True,
+                                    string='Date of reconciliation',
+                                    default='end_period_last_credit')
+    filter = fields.Char(string='Filter', size=128)
```

Comment:
You can remove size limit here. I don't see a reason to limit it. This was certainly formerly set in older version of OpenERP when it was necessary to specify this argument